tests/queries/0_stateless/01246_finalize_aggregation_race.sql

```diff
@@ -0,0 +1,23 @@
+drop table if exists test_quantile;
+create table test_quantile (x AggregateFunction(quantileTiming(0.2), UInt64)) engine = Memory;
+insert into test_quantile select medianTimingState(.2)(number) from (select * from numbers(1000) order by number desc);
+select y from (
+select finalizeAggregation(x) as y from test_quantile union all 
```

Comment:
Oh, so the race is between concurrent `convertToValue` calls that share the same source column? Like, it modifies the column despite the fact that it's constant, and turns out that it's shared. So the progressive steps to improvement would be:
1) assert that the column is not shared
2) add column mutation -- this is what you're doing in this PR
But this change would not require adding finalization to interface?

Reply:
Yes, this concrete issue may be solved this way.
However, it's wired for me, that some aggregate states now change itself at the time we get aggregation result. It looks like reading operation, and race I found is just an consequence of this behavior.
This pr was not made to fix this race. In the opposite, this race was found because I expected it may happen in current approach.
Finalization is a logical step itself, and is needed to merge pr with clusterisation.

Reply:
OK, so this is a bug fix + interface for new PR. Do you mean this one: https://github.com/ClickHouse/ClickHouse/pull/6185 ?
I'll take a look at it, maybe it'll help me understand what's going on.